src/components/badges/BadgeBlock.tsx

```diff
@@ -23,14 +23,14 @@ function Badge({
   onMinted?: (minted?: MintedToken[]) => void
   onMintFailed?: (minted?: MintedToken[]) => void
 }) {
-  const { account } = useSnapshot(WalletStore)
-  const [loading, setLoading] = useState(false)
+  const { account, mintLoading } = useSnapshot(WalletStore)
+  const [localLoading, setLocalLoading] = useState(false)
 
   const isEmailProof = proof instanceof EmailProof
 
   const checkProofAndMint = async () => {
-    setLoading(true)
-
+    WalletStore.mintLoading = true
+    setLocalLoading(true)
```

Comment:
`Local Loading` looks weird a little. Maybe it's better to call it `isBadgeMinting` or something like that?

Reply:
Maybe let's leave `loading`, like it was